spec/filters/document_provider_spec.rb

```diff
@@ -0,0 +1,234 @@
+
+    let(:document_type) { 'review' }
+
+    it 'returns all documents of a given type' do
+      expect(subject.size).to eq(1)
+      expect(subject).to match_array([review_page])
+    end
+  end
+
+  describe 'filter_by_keyword' do
+    context 'when a keyword is provided' do
+      context 'searching the title' do
+        context 'and the keyword is equal to the page title' do
+          let!(:page_entitled_keyword) { create(:insight_page_titled_annuities, site: site, layout: insight_layout) }
+          let!(:page_not_entitled_keyword) { create(:page, site: site, layout: insight_layout) }
+          let(:keyword) { 'Annuities' }
+
+          it 'returns documents where the keyword is the document title' do
+            expect(subject.size).to eq(1)
+            expect(subject).to match_array([page_entitled_keyword])
+          end
+        end
+
+        context 'and the keyword is found in the middle of the title' do
+          let!(:page_entitled_keyword) { create(:page_abt_debt_and_stress, site: site, layout: insight_layout) }
+          let!(:page_without_keyword) { create(:page, site: site, layout: insight_layout) }
+          let(:keyword) { 'stress' }
+
+          it 'returns documents with the keyword in the title' do
+            expect(subject.count).to eq(1)
+            expect(subject).to match_array([page_entitled_keyword])
+          end
+        end
+
+        context 'and the keyword is found at the start of the title' do
+          let!(:page_entitled_keyword) { create(:page_abt_debt_and_stress, site: site, layout: insight_layout) }
+          let!(:page_without_keyword) { create(:page, site: site, layout: insight_layout) }
+          let(:keyword) { 'debt' }
+
+          it 'returns documents with the keyword in the title' do
+            expect(subject.count).to eq(1)
+            expect(subject).to match_array([page_entitled_keyword])
+          end
+        end
+      end
+
+      context 'searching the content' do
+        context 'and the keyword is found in an "overview" block' do
+          let!(:page_with_keyword) { create(:insight_page_with_overview_block, site: site, layout: insight_layout) }
+          let!(:page_without_keyword) { create(:page, site: site, layout: insight_layout) }
+          let(:keyword) { 'redundancy' }
+
+          it 'returns documents which contain the keyword in an overview block' do
+            expect(subject.size).to eq(1)
+            expect(subject).to match_array([page_with_keyword])
+          end
+        end
+
+         context 'and the keyword is found in a "content" block' do
+          let!(:page_with_keyword) { create(:insight_page_with_overview_block, site: site, layout: insight_layout) }
+          let!(:page_without_keyword) { create(:page, site: site, layout: insight_layout) }
+          let(:keyword) { 'redundancy' }
+
+          it 'returns documents which contain the keyword in an content block' do
```

Comment:
in `a` content block. 

I'd also consider rewording this and/or the context block? wdyt? same for the overview block content.
`and the keyword is found in a "content" block`
`returns documents which contain the keyword in an content block`

not a blocker / don't have to change the wording, it just seems long winded.

Reply:
@willhall88 If the `describe`, `context` and `it` strings are read as a sentence then it would make sense to change the `it` string to something like `returns only those documents`, wdyt? otherwise what do you suggest?

Reply:
yeah, that would be how I would read it, and how it will appear when running rspec in documentation format. I'd be happy with that, but not a blocker if you think it should be kept as it is